docs/what.md

```diff
@@ -0,0 +1,16 @@
+# All about controllers
+
+If you've come here from the [README](../README.md), you already know that this repo holds controllers. But what even _is_ a controller?
+
+The short answer is that it's our solution to managing and disseminating state changes within our products.
+
+Every application needs to manage state. Some state is temporary. Perhaps the
+user is on a screen which keeps track of whether a button is clickable/tappable depending on whether a request is taking place; but as soon as the user leaves that screen, that information is discarded. But some state is more permanent. Perhaps a screen needs to reuse information that was gathered on a previous screen, in which case the state is cached in memory; or perhaps the entire application needs to load all of the user's data when it starts, in which case the state is persisted to disk. Either way, we need a place to store that state, and that's where controllers come into play.
+
+Currently, there are two styles of creating controllers which we've dubbed [BaseController v1](../packages/base-controller/src/BaseController.ts) and [BaseController v2](../packages/base-controller/src/BaseControllerV2.ts). Both versions offer the same set of basic functionality:
```

Comment:
I know it's already prevalent but IMO we (see what I did there) should avoid speaking from  first-person in docs like this - reserve that for discussion and P2P comms.

```suggestion
There are currently two styles of creating controllers, each with an associated TypeScript base class: [BaseController v1](../packages/base-controller/src/BaseController.ts) and [BaseController v2](../packages/base-controller/src/BaseControllerV2.ts). Both versions offer the same set of basic functionality:
```

Reply:
I can probably guess the reasoning for this, so I don't disagree, but I'm just curious — why in your opinion should this be avoided?

Reply:
There is more to it but here is one aspect: By introducing the "we", the reader is now forced to consider the author explicitly; who is writing this, when? In case of some open-source projects: Is this a message from the original author, the intermediary repository maintainers, or the company now owning the codebase? Regardless of if the reader thinks this matters or not, the context is now more dynamic. Technical documentation relates to the subject, independently of people and organizations.

For example, if we consider an alternative reality where we're documenting a proprietary hosted HTTP API, documenting rate-limiting behavior - this is one example of when I think a "we" could be warranted, depending on style.

Does that make sense?

Reply:
Yeah! I hadn't thought about that but it makes perfect sense.